### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by Mushroom-man**

### CODE REVIEW

File: src/DataBaseManager.php [hunk 1]
```diff
@@ -64,13 +65,16 @@ class DataBaseManager
...
     /**
-     * DataBase constructor.
+     * DataBaseManager constructor.
      */
     public function __construct()
     {
-        $dsn = "mysql:host=127.0.0.1;dbname=MyTest;charset=utf8";
-        $user = 'root';
-        $pass = 'LizardKing7';
+        $className = 'ConfigParser';
```
Comment (fatfattony) [inline src/DataBaseManager.php]: nahoa?

File: src/DataBaseManager.php [hunk 2]
```diff
@@ -64,13 +65,16 @@ class DataBaseManager
...
+        $className::parsingData();
+        $dsn = "mysql:host=" . ConfigParser::$parsedData["db_host:"] . ";dbname=" . ConfigParser::$parsedData["db_name:"] . ";charset=" . ConfigParser::$parsedData["charset:"];
```
Comment (fatfattony) [inline src/DataBaseManager.php]: wrong key name with ":"

File: src/ConfigParser.php [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+{
+    /**
+     * @var
+     */
+    public static $parsedData;
+
+    /**
+     *
+     */
+    public static function parsingData()
```
Comment (fatfattony) [inline src/ConfigParser.php]: with name "parsing" you can start career as senior 1s developer

File: src/DataBaseManager.php [hunk 4]
```diff
@@ -246,22 +250,64 @@ public function getQuery()
     {
         switch ($this->requestType) {
             case self::SELECT_TYPE:
-                 $this->stmt = 'SELECT ' . implode(', ', $this->fields) . ' FROM ' . implode(', ', $this->table) . ' WHERE ' .  implode(' AND ', $this->where) . ' ORDER BY ' . implode(', ', $this->orderBy);
+                 $this->getSelect();
```
Comment (fatfattony) [inline src/DataBaseManager.php]: really? do you get select here? do you user returned result?
generateSelectQuery -- this is what you method execute

File: src/ConfigParser.php [hunk 5]
```diff
@@ -0,0 +1,26 @@
+<?php
+
+class ConfigParser
+{
+    /**
+     * @var
+     */
+    public static $parsedData;
+
+    /**
```
Comment (fatfattony) [inline src/ConfigParser.php]: amazing annotation, it explain everything

File: src/ConfigParser.php [hunk 6]
```diff
@@ -0,0 +1,26 @@
+<?php
+
+class ConfigParser
+{
+    /**
+     * @var
```
Comment (fatfattony) [inline src/ConfigParser.php]: var what? some var without type... just excellent...

**Comment by fatfattony** [comment]
Attach example of config in comments plz

**Comment by Mushroom-man** [comment]
[config.txt](https://github.com/Mushroom-man/DataBaseConnector/files/3177341/config.txt)

**Pull request merged**